memestra/nbmemestra.py

```diff
@@ -4,9 +4,13 @@
 import nbformat
 
 
-def nbmemestra(nbfile, decorator):
+def nbmemestra(nb_or_file, decorator):
+    if isinstance(nb_or_file, nbformat.NotebookNode):
```

Comment:
I'd rather avoid that kind of checks and accept a single kind of node, is it possible to update the call  site instead?

Reply:
I know, but the preprocessor framework (see https://nbconvert.readthedocs.io/en/latest/nbconvert_library.html#Custom-Preprocessors) forces me to have a `preprocess` method that accepts an object of class `NotebookNode`, in which I call `nbmemestra`. The other option is to have `nbmemestra` accept a file and call a `nbmemestra_from_nbnode` or something like that, what do you think?